packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -1192,7 +1192,7 @@ export interface ChatAdapterSubscribers {
 export interface ChatAdapterThreadManagement {
     deleteMessage(messageId: string): Promise<void>;
     // (undocumented)
-    downloadAuthenticatedAttachment?: (attachmentUrl: string) => Promise<AttachmentDownloadResult>;
+    downloadAuthenticatedAttachment?: (attachmentUrl: string) => Promise<AttachmentDownloadResult | undefined>;
```

Comment:
I like the e2e implementation a lot, this api part is the only one I have concerns 

The attachmentURL seems not related to any part of chat - if you give me a random url, I can also download it for you, and it's hard for some other contoso developers to understand what is attachmentURL, and where to get it 

What I am thinking is maybe we could name it
downloadAttachment(messageId: string, index?: number) => Promise<AttachmentDownloadResult | undefined>; (they put multiple blobsUrl in the result)

Then the function itself could figure out what to download

I've still approved the PR since you are bug bashing it, let's see if we could make this happen if we haven't done ARB yet, or we could mark this as an item for GA

Reply:
And for naming - it feels it gets a url then return a blobUrl, so it could be just downloadAttachment no matter it is authenticated or not, just because we are now only using it for authenticatedAttachment doesn't mean it is only designed for non-authenticated one (maybe an option could be involved in the future)